rules/linux-hardening/raw.rego

```diff
@@ -1,111 +1,80 @@
 }
 
-# Fails if workload does not define linux security hardening 
-deny[msga] {
-    wl := input[_]
-	spec_template_spec_patterns := {"Deployment","ReplicaSet","DaemonSet","StatefulSet","Job"}
-	spec_template_spec_patterns[wl.kind]
-    is_unsafe_workload(wl)
-    container := wl.spec.template.spec.containers[i]
-    is_unsafe_container(container)
-
-	fixPaths := [
-	{"path": sprintf("spec.template.spec.containers[%v].seccompProfile", [format_int(i, 10)]), "value": "YOUR_VALUE"},
-	{"path": sprintf("spec.template.spec.containers[%v].seLinuxOptions", [format_int(i, 10)]), "value": "YOUR_VALUE"},
-	{"path": sprintf("spec.template.spec.containers[%v].capabilities.drop", [format_int(i, 10)]), "value": "YOUR_VALUE"}
-	]
-
-
-	msga := {
-		"alertMessage": sprintf("Workload: %v does not define any linux security hardening", [wl.metadata.name]),
-		"packagename": "armo_builtins",
-		"alertScore": 7,
-		"failedPaths": [],
-		"fixPaths": fixPaths,
-		"alertObject": {
-			"k8sApiObjects": [wl]
-		}
-	}
+is_unsafe_obj(obj) := fix_paths {
+	obj.kind == "Pod"
+	fix_paths := are_unsafe_specs(obj, ["spec"], ["metadata", "annotations"])
+} else := fix_paths {
+	obj.kind == "CronJob"
+	fix_paths := are_unsafe_specs(obj, ["spec", "jobTemplate", "spec", "template", "spec"], ["spec", "jobTemplate", "spec", "template", "metadata", "annotations"])
+} else := fix_paths {
+	obj.kind in ["Deployment", "ReplicaSet", "DaemonSet", "StatefulSet", "Job"]
+	fix_paths := are_unsafe_specs(obj, ["spec", "template", "spec"], ["spec", "template", "metadata", "annotations"])
 }
 
 
-# Fails if pod does not define linux security hardening 
-deny[msga] {
-	wl := input[_]
-	wl.kind == "CronJob"
-    is_unsafe_cronjob(wl)
-	container = wl.spec.jobTemplate.spec.template.spec.containers[i]
-    is_unsafe_container(container)
-
-	fixPaths := [
-	{"path": sprintf("spec.jobTemplate.spec.template.spec.containers[%v].seccompProfile", [format_int(i, 10)]), "value": "YOUR_VALUE"},
-	{"path": sprintf("spec.jobTemplate.spec.template.spec.containers[%v].seLinuxOptions", [format_int(i, 10)]), "value": "YOUR_VALUE"},
-	{"path": sprintf("spec.jobTemplate.spec.template.spec.containers[%v].capabilities.drop", [format_int(i, 10)]), "value": "YOUR_VALUE"}
+# Do we need here anotation_path, or it just
+are_unsafe_specs(obj, specs_path, anotation_path) := paths {
+	# spec
+	specs := object.get(obj, specs_path, null)
+	specs != null
+	are_seccomp_and_selinux_disabled(specs)
+
+	# annotation
+	annotations := object.get(obj, anotation_path, [])
+	app_armor_annotations := [annotations[i] | annotation = i; startswith(i, "container.apparmor.security.beta.kubernetes.io")]
+	count(app_armor_annotations) == 0
+
+	# container
+	containers_path := array.concat(specs_path, ["containers"])
+	containers := object.get(obj, containers_path, [])
+
+	# Psuedo code explanation:
+	# for i, container in containers
+	#  		if is_unsafe_container:
+	# 			fix_paths += [(containers_path[i] + fix_fields) for j, field in fix_fields]
```

Comment:
did you mean `fix_paths += [(containers_path[i] + field) for j, field in fix_fields]` ?

Reply:
Probably :)